### PR TITLE
Implement borland delphi demangler and fix gpl build

### PR DESCRIPTION
**Pull request by wargio**
# DO NOT SQUASH

I have made a big refactoring on how the tests are executed, so don't worry too much for the changes on the other files.

I have added tests and the demangler for borderland delphi (c++)

### CODE REVIEW

**Review by XVilka — CHANGES_REQUESTED** [review]
What is "borderland delphi"? I thought it's "Borland"
Also, could you please extract "minunit.h" changes into a separate commit, so the possible synchronization with mainstream miunit or Rizin one will be easier?

**Comment by wargio** [comment]
no, i prefer to keep this `minunit.h` as special, since it is dedicated for demangling. let me try to split the PR into 2 commits.

File: src/cxx.c [hunk 1]
```diff
@@ -1,10 +1,12 @@
 // SPDX-FileCopyrightText: 2013-2019 pancake <pancake@nopcode.org>
 // SPDX-License-Identifier: LGPL-3.0-only
 #include "demangler_util.h"
-#include "cxx/demangle.h"
+#include "borland.h"
```
Comment (XVilka) [inline src/cxx.c:4]: I think this change doesn't belong to this commit and should be moved to the next one?

File: src/cxx.c [hunk 2]
```diff
@@ -86,3 +88,15 @@ char *libdemangle_handler_cxx(const char *str) {
 	return out;
 }
 #endif
+
+char *libdemangle_handler_cxx(const char *symbol) {
+	char *result = demangle_borland_delphi(symbol);
```
Comment (XVilka) [inline src/cxx.c:93]: Same

**Review by XVilka — CHANGES_REQUESTED** [review]
LGTM with minor nitpick

File: src/cxx.c (hunk 1 above)
Comment (wargio) [reply src/cxx.c:4]: fixed.

File: test/test_borland.c [hunk 3]
```diff
@@ -0,0 +1,185 @@
+// SPDX-FileCopyrightText: 2021 deroad <wargio@libero.it>
```
Comment (XVilka) [inline test/test_borland.c:1]: 2023?

**Review by XVilka — APPROVED** [review]

File: test/test_borland.c (hunk 3 above)
Comment (wargio) [reply test/test_borland.c]: Done. could you approve?

**Review by XVilka — APPROVED** [review]
I already did
